dotCMS/src/main/java/com/dotcms/util/FileSplitter.java

```diff
@@ -0,0 +1,191 @@
+        private int               cursor;
+        private final InputStream stream;
+        private final long        fileLength;
+        private final byte[]      chuckbuffer;
+
+        public FileSplitterIterator() {
+
+            this.index       = 0;
+            this.cursor      = 0;
+            this.fileLength  = file.length();
+            this.stream      = Try.of(()->com.liferay.util.FileUtil.createInputStream
+                    (file.toPath(), "none")).getOrElseThrow(DotRuntimeException::new);
+            this.chuckbuffer = new byte[FileSplitter.this.chuckSize];
+        }
+
+        @Override
+        public boolean hasNext() {
+            return this.cursor < this.fileLength;
+        }
+
+        @Override
+        public File next() {
+
+            if (this.hasNext()) {
+                // check byte count from cursor...
+                final int readBytes  = Try.of(()->this.stream.read(this.chuckbuffer)).getOrElseThrow(DotRuntimeException::new);
+                final File chuckFile = FileSplitter.this.createSubFileFunction.apply(file.getName(), this.index);  // this file should have the chuck in the file and readBytes
+                try (OutputStream outputStream = FileSplitter.this.outputStreamCreatorFunction.apply(chuckFile)) {
+
+                    outputStream.write(this.chuckbuffer, 0, readBytes);
+                    outputStream.flush();
+                } catch (IOException e) {
+
+                    throw new DotRuntimeException(e);
+                }
+
+                this.cursor         += readBytes;
+                this.index++;
+
+                return chuckFile;
+            }
+
+            throw new NoSuchElementException();
+        }
+
+        @Override
+        public void close() throws IOException {
+            if (null != this.stream) {
+                this.stream.close();
+            }
+        }
+    }
+
+    public static void main(String[] args) {
+
+        List<File> files = new ArrayList<>();
+        File file = new File ("/Users/jsanca/Downloads/[JAVA][Beginning Java 8 Games Development].pdf" );
+
+        try (FileSplitter fileSplitter = new FileSplitter(file, 1048576, // 1 mb
+                (filename, index)-> new File("/Users/jsanca/Downloads/tmp/" + index + "-" + filename + ".bin"))) {
+
+            for (final File partitionFile : fileSplitter) {
+
+                files.add(partitionFile);
+                System.out.println(partitionFile.getAbsolutePath());
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+
+        try (final FileJoiner fileJoiner = new FileJoiner(new File("/Users/jsanca/Downloads/tmp/ensambled.pdf"))){
+            for (final File chuckFile : files) {
+
+                fileJoiner.join(chuckFile);
+                fileJoiner.flush();
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+
+        System.out.println("DONE");
+        System.exit(0);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [System.exit() should not be used in J2EE/JEE apps](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5547873)